module4/homework/appendNewRecipe/recipeAppend.cpp

```diff
@@ -0,0 +1,55 @@
+#include <iostream>
+
+#include "recipeAppend.hpp"
+
+bool appendNewRecipe(const std::string& filename,
+                     const std::vector<std::string>& formattedIngredients,
+                     const std::stringstream& formattedRecipe) {
+    std::fstream recipe(filename, recipe.out | recipe.app);
+    if (recipe.is_open()) {
+        for(const auto& el : formattedIngredients) {
+            recipe << el << '\n';
+        }
+        recipe << formattedRecipe.rdbuf() << '\n';
+        recipe.close();
+    } else {
+        std::cerr << "File open error!" << std::endl;
+        return false;
+    }
+    return true;
+}
+
+std::vector<std::string> formatIngredients(const std::list<std::string>& ingredients,
+                                           const std::deque<std::pair<size_t, char>>& amount) {
+    std::vector<std::string> formattedIngredients;
+    
+    formattedIngredients.emplace_back("Skladniki:");
```

Comment:
Generally we should avoid using magic values in the code. It's better to make "Skladniki:" const variable. Then when you will be working in larger projects that values would easier to find and change.

Reply:
+1

Reply:
Good point. Thanks!

Reply:
+1